Inspicio/Controllers/ImagesController.cs

```diff
@@ -152,8 +152,20 @@ public class CommentInfo
             return View(FullReviewData);
         }
 
-
-
+       
+        public JsonResult GetComments(int? Id)
+        {
+            List<FullReviewData.CommentInfo> comments = new List<FullReviewData.CommentInfo>();
+            var AllComments = _context.Comments.Where(c => c.ImageId == Id);
+            foreach(Comment SingleComment in AllComments)
+            {
+                var CommentInfo = new FullReviewData.CommentInfo();
+                CommentInfo.comment = SingleComment;
+                CommentInfo.PosterProfileName = _context.Users.Where(u => u.Id == SingleComment.OwnerId).Select(u => u.ProfileName).SingleOrDefault();
```

Comment:
Should `PosterProfileName` ever be missing? If not, then I'd got with `Single()` instead of `SingleOrDefault()` that way it'll show you an error close to the problem, rather than as a missing name and then wondering what's gone wrong. In general, when something has gone wrong it can be handy to break straight away in a noticeable way, to make it easy to fix.